library/core/src/slice/index.rs

```diff
@@ -392,7 +391,7 @@ unsafe impl<T> SliceIndex<[T]> for ops::Range<usize> {
         );
         // SAFETY: see comments for `get_unchecked` above.
         unsafe {
-            let new_len = unchecked_sub(self.end, self.start);
+            let new_len = self.end.unchecked_sub(self.start);
```

Comment:
Fair, although we're ultimately going to have a bunch of checks duplicated when we have unsafe preconditions floating around *anyway*.

And besides, in a correct code base, the precondition checks will *always* be duplicated, since you should be only using them when they've been checked ahead of time anyway. It's just a matter of how many times they're duplicated.